DEVELOPER.md

```diff
@@ -0,0 +1,43 @@
+# Notes for Development
+
+## Development Environment
+
+## Building from Source
+
+To build and test the driver, the standard cargo commands can be used from the root directory.
+
+For an unoptimized build with debugging information (most common):
+```
+cargo build
+```
+
+For an optimized build with debugging info (mac)
+```
+cargo build --features odbc-sys/iodbc,cstr/utf32 --profile=release-with-debug
+```
+And for an optimized build with debugging info (windows, linux)
+```
+cargo build --profile=release-with-debug
+```
+The resulting build files will be output to the `target` directory in the `debug` or `release` folders respectively.
+
+## Running Tests
+
+### To run unit tests
+
+Similar to building, standard cargo commands can be used here:
+
+```
+cargo test unit
+```
+
+### Other types of tests
+The other tests that are run are integration and result set tests. These involve more setup, and that setup is largely operating system specific. For more information on how to run these, consider checking [evergreen.yml](evergreen.yml), and following the steps taken for the given operating system-task (i.e. `macos-integration-test` or `windows-result-set-test`)
```

Comment:
One potential issue here is that `go` is set up differently on a local machine, and on windows especially it's possible to run the integration tests without cygwin by using wsl to run adf, so you only have to worry about the integration tests.

Reply:
That makes sense, and thanks for calling out windows (I have only had the chance to run on windows hosts, which === evergreen). I can just hedge the language here so it is just nudging in the right direction, or add more specific instructions for windows/mac (I would probably need a little help with local windows in that case). What do you think is the best way to proceed?

Reply:
I can contribute what I do on Windows if you want. It's actually pretty easy. I run WSL and run adf there (so linux setup), then on windows it "just connects" when I run the tests. The only thing is to make sure environment variables are properly set.